### ISSUE TITLE
Missing comma in list values lint

### ISSUE THREAD

**Issue by VFLashM** [PyCQA/pylint#1589]
### Steps to reproduce

```python
DATA = [
    'a',
    'b' # missing comma
    'c',
    'd',
]
```

### Current behavior

No warning

### Expected behavior

It should probably be reported.
String values get silently concatenated because of missing comma. This is most certainly not what user wanted or expected.

### pylint --version output

pylint 1.7.2, 
astroid 1.5.3
Python 2.7.12 (default, Sep 20 2016, 14:05:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]

**Comment by PCManticore** [comment]
The problem with this check is that it can result in a lot of false positives. How should you figure out that the string concatenation is not implicitly required? We also use this pattern a lot in our codebase for example and this check will emit on those instances as if they are problematic, which in fact is not.

**Comment by VFLashM** [comment]
Do you have such implicit string concatenation in list values or just somewhere in code?
I'm talking specifically about lists.

I can argue that it would be not a false positive, but a code smell.
For example, single character variables are not problematic on their own, but they are linted nonetheless.

**Comment by PCManticore** [comment]
In tuples. But the principle remains. Why should we warn about lists and not about tuples?

**Comment by VFLashM** [comment]
Yes, it stands for tuples too. I meant that stray values are more or less safe.
